go/logic/migrator.go

```diff
@@ -1279,30 +1279,31 @@ func (this *Migrator) executeWriteFuncs() error {
 					return err
 				}
 			}
-		default:
-			{
+		case copyRowsFunc := <-this.copyRowsQueue:
+		priority:
+			for {
 				select {
-				case copyRowsFunc := <-this.copyRowsQueue:
+				case eventStruct := <-this.applyEventsQueue:
 					{
-						copyRowsStartTime := time.Now()
-						// Retries are handled within the copyRowsFunc
-						if err := copyRowsFunc(); err != nil {
-							return this.migrationContext.Log.Errore(err)
-						}
-						if niceRatio := this.migrationContext.GetNiceRatio(); niceRatio > 0 {
-							copyRowsDuration := time.Since(copyRowsStartTime)
-							sleepTimeNanosecondFloat64 := niceRatio * float64(copyRowsDuration.Nanoseconds())
-							sleepTime := time.Duration(int64(sleepTimeNanosecondFloat64)) * time.Nanosecond
-							time.Sleep(sleepTime)
+						if err := this.onApplyEventStruct(eventStruct); err != nil {
+							return err
 						}
 					}
 				default:
-					{
-						// Hmmmmm... nothing in the queue; no events, but also no row copy.
-						// This is possible upon load. Let's just sleep it over.
-						this.migrationContext.Log.Debugf("Getting nothing in the write queue. Sleeping...")
-						time.Sleep(time.Second)
-					}
+					break priority
+				}
+			}
+			{
+				copyRowsStartTime := time.Now()
+				// Retries are handled within the copyRowsFunc
+				if err := copyRowsFunc(); err != nil {
+					return this.migrationContext.Log.Errore(err)
+				}
+				if niceRatio := this.migrationContext.GetNiceRatio(); niceRatio > 0 {
+					copyRowsDuration := time.Since(copyRowsStartTime)
+					sleepTimeNanosecondFloat64 := niceRatio * float64(copyRowsDuration.Nanoseconds())
+					sleepTime := time.Duration(time.Duration(int64(sleepTimeNanosecondFloat64)) * time.Nanosecond)
```

Comment:
```suggestion
					sleepTime := time.Duration(time.Duration(sleepTimeNanosecondFloat64) * time.Nanosecond)
```
The `int64()` conversion is unnecessary and is making the linter upset:
```
unnecessary conversion (unconvert)
```

Reply:
yeah, this is a good idea

Reply:
the outer time.Duration is unnecessary too